Description: Set nomodeset for recovery mode
 Also disable 'set gfxpayload=keep' even if the system normally supports it.
 .
 https://launchpad.net/ubuntu/+spec/desktop-o-xorg-tools-and-processes
Author: Colin Watson <cjwatson@ubuntu.com>
Forwarded: no
Last-Update: 2013-01-21

Index: b/util/grub.d/10_linux.in
===================================================================
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -130,7 +130,9 @@
 	  echo "	load_video" | sed "s/^/$submenu_indentation/"
       fi
   fi
-  echo "	gfxmode \$linux_gfx_mode" | sed "s/^/$submenu_indentation/"
+  if [ x$type != xrecovery ] ; then
+      echo "	gfxmode \$linux_gfx_mode" | sed "s/^/$submenu_indentation/"
+  fi
 
   echo "	insmod gzio" | sed "s/^/$submenu_indentation/"
 
@@ -305,10 +307,10 @@
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
     if [ -x /lib/recovery-mode/recovery-menu ]; then
       linux_entry "${OS}" "${version}" recovery \
-                  "recovery ${GRUB_CMDLINE_LINUX}"
+                  "recovery nomodeset ${GRUB_CMDLINE_LINUX}"
     else
       linux_entry "${OS}" "${version}" recovery \
-                  "single ${GRUB_CMDLINE_LINUX}"
+                  "single nomodeset ${GRUB_CMDLINE_LINUX}"
     fi
   fi
 
